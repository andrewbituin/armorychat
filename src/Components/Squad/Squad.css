.squad-container {
  position: relative;
  display: flex;
  height: 118px;
  border: 1px solid transparent;
}

.squad-container:hover, .squad-container:focus {
  background-color: #3b3e48 !important;
  border-radius: 0;
  border: 1px solid #6e6f72;
  box-shadow: 0 5px 13px 0 #00000080;
  cursor: pointer;
  z-index: 1;
}

.squad-list {
  margin-top: 124px;
  padding: 0 30px;
  list-style: none;
}

.squad-container {
  margin: 0 -11px;
  border-radius: 5px;
  padding: 0 11px;
}

.squad-container:nth-child(even) {
  background-color: #2c2e35;
}

.squad-list h4 {
  min-width: 160px;
  color: #7faf23;
  font-size: 18px;
  font-weight: 600;
  margin: 0 10px 0 0;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;

}

.squad-list p, .squad-list span, .squad__description {
  margin: 0;
  font-weight: normal;
  display: inline-block;
  white-space: nowrap;
}

.squad-list span {
  margin-right: 10px;
}

.squad__description {
  position: relative;
  color: #b3b3b3;
  overflow: hidden;
  text-overflow: ellipsis;
  width: 80%;
}

.squad__details-bottom {
  display: flex;
  align-items: center;
}

.squad__gamemode-image {
  width: 60px;
  height: 60px;
  box-shadow: 0 1px 8px 0 #00000080;
  border-radius: 5px;
  border: 1px solid #909193;
}

.squad__gamemode-image-icon {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #565d69;
  font-size: 50px;
}

.squad__details {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: calc(100% - 200px);
}

.squad__details-top {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

.squad__spots-list {
  width: 320px;
  list-style: none;
  padding: 0;
}

.squad__spots-list li {
  display: inline-block;
}

.squad__spots-list li:not(:last-child) {
  margin-right: 4px;
}

.squad__spots-full {
  position: absolute;
  top: -7px;
  left: -7px;
  width: 44px;
  height: 50px;
}

.squad__spots-filled, .squad__spots-empty{
  display: flex;
  position: absolute;
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
}

.squad__spots-empty {
  background-color: #565d69;
}

.squad__spots-filled {
  background-color: #2c2e35b9;
  color: #fff !important;
}

.squad__spots-image-container {
  position: relative;
  width: 32px;
  height: 38px;
  overflow: hidden;
  border-radius: 5px;
  box-shadow: 0 1px 8px 0 #00000080;
  border: 1px solid #909193;
}

.squad__spots-first-half, .squad__spots-second-half {
  position: absolute;
  display: inline-block;
  font-size: 0;
  width: 44px;
  height: 50px;
  top: -7px;
}

.squad__spots-first-half {
  clip-path: inset(6px 22px 6px 6px);
  top: -7px;
  left: -7px;
}

.squad__spots-second-half {
  clip-path: inset(6px 6px 6px 22px);
  top: -7px;
  right: -7px;
}

.squad__gamemode-image-container {
  display: flex;
  align-items: center;
  padding: 0 10px;
  width: 80px;
}

.squad__requirements {
  display: flex;
  align-items: center;
  font-weight: normal;
  width: 140px;
}

.squad__avatar-image {
  position: absolute;
  left: -4px;
  width: 38px;
  height: 38px;
}

.squad__owner {
  display: flex;
  align-items: center;
  width: 200px;
  padding-bottom: 5px;
}

.squad__owner p {
  color: #fff;
}

.squad__owner div {
  margin-right: 10px;
}