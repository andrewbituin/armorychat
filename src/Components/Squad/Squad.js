import React from 'react';
import Spot from '../Spot/Spot';
import './Squad.css';
import config from '../../config';


export default function Squad(props) {
  function generateSpots(party) {
    return party.spots
      .map((spot, index) => {
        return (
          <Spot key={index} index={index} spot={spot} gameId={props.gameId} partyId={party.id} history={props.history}/>
        );
      });
  }

  function generateRequirements() {
    return props.party.reqs.map((req, i) => {
      if (req.name) {
        return <span className="small-detail" key={i}>{req.name}</span>
      }
      return null;
    });
  }

  return (
    <li className="squad-container" aria-label='Open Squad Details' tabIndex="0" onClick={() => props.onClick(props.index)}>
      <div className="squad__gamemode-image-container">
        {(props.party.gamemode) 
          ? <img className="squad__gamemode-image" src={`${config.IMAGES_ENDPOINT}/${props.gameId}/${props.party.gamemode.icon_url}`} alt=""/>
          : <i className="squad__gamemode-image fas fa-globe squad__gamemode-image-icon"></i>
        }
      </div>
      <div className="squad__details">
        <div className="squad__details-top">
          <h4>{props.party.title}</h4>
          <ul className="squad__spots-list">
            {generateSpots(props.party)}
          </ul>
          <div className="squad__owner">
            <div className="squad__spots-image-container">
              <img className="squad__avatar-image" src={`${config.IMAGES_ENDPOINT}/user-icons/${props.party.owner_id.avatar_url}`} alt=""/>
            </div>
            <p>{props.party.owner_id.username}</p>
          </div>
        </div>
        <div className="squad__details-bottom">
          {props.party.gamemode && <span className="small-detail">{props.party.gamemode.name}</span>}
          <p className="squad__description">{props.party.description || 'No description...'}</p>
        </div>
      </div>
      <div className="squad__requirements">
        {generateRequirements()}
      </div>
    </li>
  )
}