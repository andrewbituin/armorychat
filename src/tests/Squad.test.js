import React from "react";
import Squad from "../Components/Squad/Squad";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const props = {
  id: "fb1d3c63-6a72-4013-be82-5b523c1dd1cd",
  game_id: "aa0e8ce9-1a71-42e7-804d-6838556fa6ed",
  title: "Test Title",
  owner_id: 1,
  description: "This is a description of test Squad.",
  require_app: true,
  gamemode: {
      name: "Quick Play",
      icon_url: "Gamemode-Quickplay.png"
  },
  reqs: [
      {
          name: "Diamond"
      },
      {
          name: "Bronze"
      }
  ],
  spots: [
      {
          id: "64ed5ba8-78db-44c6-ae60-46e6a2a07ff9",
          filled: null,
          roles: [
              {
                  name: "Genji",
                  icon_url: "Icon-Genji.png"
              },
              {
                  name: "Support",
                  icon_url: "Icon-Support.png"
              }
          ]
      },
      {
          id: "25539899-aae0-469e-92c1-a2116badc84c",
          filled: {
              username: "admin",
              avatar_url: "https://i.ebayimg.com/images/g/PfAAAOSwA3dYIPRN/s-l300.jpg"
          },
          roles: [
              {}
          ]
      }
  ]
};

describe("Squad component", () => {
  it("renders without crashing", () => {
    shallow(<Squad party={props} />);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Squad party={props} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
