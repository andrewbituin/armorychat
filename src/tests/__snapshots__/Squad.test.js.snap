// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Squad component renders the UI as expected 1`] = `
<li
  aria-label="Open Squad Details"
  className="squad-container"
  onClick={[Function]}
  tabIndex="0"
>
  <div
    className="squad__gamemode-image-container"
  >
    <img
      alt=""
      className="squad__gamemode-image"
      src="https://limitless-brushlands-45977.herokuapp.com/images/undefined/Gamemode-Quickplay.png"
    />
  </div>
  <div
    className="squad__details"
  >
    <div
      className="squad__details-top"
    >
      <h4>
        Test Title
      </h4>
      <ul
        className="squad__spots-list"
      >
        <li
          className="squad__spots-image-container"
        >
          <img
            alt=""
            className="squad__spots-first-half"
            src="https://limitless-brushlands-45977.herokuapp.com/images/undefined/Icon-Genji.png"
          />
          <img
            alt=""
            className="squad__spots-second-half"
            src="https://limitless-brushlands-45977.herokuapp.com/images/undefined/Icon-Support.png"
          />
        </li>
        <li
          className="squad__spots-image-container"
        >
          <i
            className="squad__spots-empty fas fa-users"
          />
          <i
            className=" squad__spots-filled fas fa-check"
          />
        </li>
      </ul>
      <div
        className="squad__owner"
      >
        <div
          className="squad__spots-image-container"
        >
          <img
            alt=""
            className="squad__avatar-image"
            src="https://limitless-brushlands-45977.herokuapp.com/images/user-icons/undefined"
          />
        </div>
        <p />
      </div>
    </div>
    <div
      className="squad__details-bottom"
    >
      <span
        className="small-detail"
      >
        Quick Play
      </span>
      <p
        className="squad__description"
      >
        This is a description of test Squad.
      </p>
    </div>
  </div>
  <div
    className="squad__requirements"
  >
    <span
      className="small-detail"
    >
      Diamond
    </span>
    <span
      className="small-detail"
    >
      Bronze
    </span>
  </div>
</li>
`;
